src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/SqlConnection.cs

```diff
@@ -179,6 +181,9 @@ public void RunInTransaction(Action action)
         private void Rollback(bool throwOnError)
             => this.ExecuteCommand("rollback transaction", throwOnError);
 
+        private void Configure()
+            => raw.sqlite3_exec(_handle, "pragma journal_mode=wal");
```

Comment:
❓ Are there any problems with executing this after the database is already populated? What about cases where the process opens multiple connections?

⚠️ We should test the behavior on a networked file system, where shared memory is not available.

Reply:
As I understand this is not a problem for populated databases (I did most of my testing starting with a copy of an existing non-WAL DB), and the mode is persistent after being set (can be disabled).

Concurrent writes should be faster (because of append-only nature), reads are same or single-digit-percent slower according to [documentation](https://sqlite.org/wal.html). With this specific use case, I believe that any WAL-related slowdown will be offset by not needing to open nonexistent files (the original reason for #22339 and this PR).

Working over networked file system wouldn't work (documentation explicitly states this), but the failure mode is not known for now, I'll take a look.

Reply:
My tests shows that it works over SMB (symlinked vs15 directory to a remote fileserver over SSL VPN over Internet, strangely it's still usable): I believe that SQLite documentation term for "network file system" means "something like NFS" that doesn't support SHM and/or locking.

I think that most network file systems on Mac or Linux won't work though, but running something that does *frequent* file locking over NFS or something similar is already a bad idea, as far as I know, NFS will just freeze if lock acquires/releases are in the 10^3/s range or greater, and distributed systems are **much** worse.

Reply:
With NFS the `mmap` works ok, but the mapped region is *usually* not actively synced with the host. For a single-client scenario it's actually not much different from non-SHM case, and multi-client (meaning different machines accessing the same SQLite database over NFS) is probably an edge case not worth considering in a world of DVCSes.

Reply:
Even better, the [SQLite documentation](https://www.sqlite.org/lockingv3.html#how_to_corrupt) states that "Your best defense is to not use SQLite for files on a network filesystem." even when not using WAL because of "POSIX advisory locking is known to be buggy or even unimplemented on many NFS implementations (including recent versions of Mac OS X) and that there are reports of locking problems for network filesystems under Windows".
I'm really not sure if over-the-network scenario is worth testing for WAL, even if base case (non-WAL) has recommendations against using network filesystems to store databases.